backend/src/util/validate.ts

```diff
@@ -26,6 +26,8 @@ async function calculateBalance(
 ): Promise<{ balance: Decimal; decay: Decay; lastTransactionId: number } | null> {
   const lastTransaction = await Transaction.findOne({ userId }, { order: { balanceDate: 'DESC' } })
   if (!lastTransaction) return null
+  // negative amount should not be allowed
+  if (amount.greaterThan(0)) return null
```

Comment:
Now I finally understand your point! We call `calculateBalance` twice. Once from the transaction resolver and once from the transaction link resolver. In both cases we multiply the amount by -1. We should do this operation in this function and the test here would be `amount <= 0`. Makes much more sense

Reply:
Go ahead. Get rid of the `mul(-1)` in the calls of `calculateBalance` and do it here instead.

Reply:
I also added some utility to avoid all the mess regarding the toString() methods.
Take a quick look at it, you may like it or not hahaha